interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -354,29 +356,41 @@ public void pingPong() throws Exception {
                 .setBody(ByteString.copyFrom(new byte[58979])))
             .build());
 
-    @SuppressWarnings("unchecked")
-    StreamObserver<StreamingOutputCallResponse> responseObserver = mock(StreamObserver.class);
+    final ArrayBlockingQueue<Object> queue = new ArrayBlockingQueue<Object>(5);
     StreamObserver<StreamingOutputCallRequest> requestObserver
-        = asyncStub.fullDuplexCall(responseObserver);
+        = asyncStub.fullDuplexCall(new StreamObserver<StreamingOutputCallResponse>() {
+          @Override
+          public void onNext(StreamingOutputCallResponse response) {
+            queue.add(response);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            queue.add(t);
+          }
+
+          @Override
+          public void onCompleted() {
+            queue.add("Completed");
+          }
+        });
     for (int i = 0; i < requests.size(); i++) {
+      assertNull(queue.peek());
       requestObserver.onNext(requests.get(i));
-      verify(responseObserver, timeout(operationTimeoutMillis())).onNext(goldenResponses.get(i));
-      verifyNoMoreInteractions(responseObserver);
+      assertEquals(queue.poll(operationTimeoutMillis(), TimeUnit.MILLISECONDS),
```

Comment:
Reverse the arguments to this `assertEquals()`. It should be `assertEquals(expected, actual)`.